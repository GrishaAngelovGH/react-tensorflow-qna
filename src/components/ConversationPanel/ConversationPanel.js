import "./ConversationPanel.scss"

const ConversationPanel = () => {
  return (
    <div className="conversation-panel">
      <div className="left-col">Left Column</div>
      <div className="right-col">Right Column</div>
    </div>
  )
}

export default ConversationPanel