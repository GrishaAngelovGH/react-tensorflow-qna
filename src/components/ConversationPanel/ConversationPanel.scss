.conversation-panel {
  display: flex;
  height: calc(100vh - 100px);
}

.left-col {
  width: 70%;
}

.right-col {
  width: 30%;
}

.conversation {
  height: 400px;
}