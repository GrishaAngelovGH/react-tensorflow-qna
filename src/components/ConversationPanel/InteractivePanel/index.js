import InteractivePanel from "./InteractivePanel"

export default InteractivePanel