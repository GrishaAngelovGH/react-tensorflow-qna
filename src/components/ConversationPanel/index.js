import ConversationPanel from "./ConversationPanel"

export default ConversationPanel