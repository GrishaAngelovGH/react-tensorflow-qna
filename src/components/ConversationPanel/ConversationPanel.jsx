import { useState } from "react"

import useFile from "hooks/useFile"
import useProcessQuestion from "hooks/useProcessQuestion"

import Form from "react-bootstrap/Form"
import Conversation from "./Conversation"
import InteractivePanel from "./InteractivePanel"
import Questions from "./Questions"

import "./ConversationPanel.scss"

const ConversationPanel = () => {
  const [showFileContent, setShowFileContent] = useState(true)
  const [showQuestions, setShowQuestions] = useState(false)
  const [question, setQuestion] = useState('')

  const fileContent = useFile()
  const [answer, isProcessing] = useProcessQuestion(question, fileContent)

  const handleToggleFileContent = () => {
    setShowFileContent(!showFileContent)
    setShowQuestions(false)
  }

  const handleToggleQuestions = () => {
    setShowQuestions(!showQuestions)
    setShowFileContent(false)
  }

  return (
    <div className="conversation-panel">
      <div className="left-col">
        {
          question.length > 0 && (
            <Conversation question={question} answer={answer} isProcessing={isProcessing} />
          )
        }

        <InteractivePanel
          onSendQuestion={setQuestion}
          onToggleFileContent={handleToggleFileContent}
          onToggleQuestions={handleToggleQuestions}
        />
      </div>

      <div className="right-col m-2">
        {
          showFileContent && (
            <Form.Control
              as="textarea"
              defaultValue={fileContent}
              disabled
              className="h-100"
            />
          )
        }
        {
          showQuestions && (
            <Questions onClick={setQuestion} />
          )
        }
      </div>
    </div>
  )
}

export default ConversationPanel