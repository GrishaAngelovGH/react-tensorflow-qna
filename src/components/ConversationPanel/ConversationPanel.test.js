import { render } from '@testing-library/react'

import ConversationPanel from './ConversationPanel'

test('should render ConversationPanel component', () => {
  const view = render(<ConversationPanel />)

  expect(view).toMatchSnapshot()
})