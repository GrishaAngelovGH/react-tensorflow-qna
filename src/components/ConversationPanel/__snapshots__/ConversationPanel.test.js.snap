// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConversationPanel component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="conversation-panel"
      >
        <div
          class="left-col"
        >
          <div
            class="row justify-content-center fixed-bottom ps-2"
          >
            <div
              class="col-8 col-md-4 col-lg-5"
            >
              <div
                class="mb-3 input-group"
              >
                <input
                  class="border-primary form-control"
                  placeholder="Ask a question"
                  value=""
                />
                <button
                  class="btn btn-primary bi bi-send"
                  type="button"
                />
              </div>
            </div>
            <div
              class="col-8 col-md-3 col-lg-5"
            >
              <div
                class="btn-group"
                role="group"
              >
                <button
                  class="btn btn-outline-success bi bi-layout-sidebar-reverse"
                  type="button"
                />
                <button
                  class="btn btn-outline-success bi bi-list"
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="right-col m-2"
        >
          <textarea
            class="h-100 form-control"
            disabled=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="conversation-panel"
    >
      <div
        class="left-col"
      >
        <div
          class="row justify-content-center fixed-bottom ps-2"
        >
          <div
            class="col-8 col-md-4 col-lg-5"
          >
            <div
              class="mb-3 input-group"
            >
              <input
                class="border-primary form-control"
                placeholder="Ask a question"
                value=""
              />
              <button
                class="btn btn-primary bi bi-send"
                type="button"
              />
            </div>
          </div>
          <div
            class="col-8 col-md-3 col-lg-5"
          >
            <div
              class="btn-group"
              role="group"
            >
              <button
                class="btn btn-outline-success bi bi-layout-sidebar-reverse"
                type="button"
              />
              <button
                class="btn btn-outline-success bi bi-list"
                type="button"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="right-col m-2"
      >
        <textarea
          class="h-100 form-control"
          disabled=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
